library/common/jni/jni_interface.cc

```diff
@@ -175,41 +181,47 @@ static void jvm_on_data(envoy_data data, bool end_stream, void* context) {
   // Here '0' (for which there is no named constant) indicates we want to commit the changes back
   // to the JVM and free the c array, where applicable.
   env->ReleasePrimitiveArrayCritical(j_data, critical_data, 0);
-  env->CallVoidMethod(j_context, jmid_onData, j_data, end_stream ? JNI_TRUE : JNI_FALSE);
+  jobject result =
+      env->CallObjectMethod(j_context, jmid_onData, j_data, end_stream ? JNI_TRUE : JNI_FALSE);
 
   data.release(data.context);
   env->DeleteLocalRef(j_data);
   env->DeleteLocalRef(jcls_JvmCallbackContext);
+  return result;
 }
 
-static void jvm_on_metadata(envoy_headers metadata, void* context) {
+static void* jvm_on_metadata(envoy_headers metadata, void* context) {
```

Comment:
This isn't related to this PR but do we want to remove the on metadata callbacks since we are not using them or exposing them?